portal/experimentstool/templates/experimentstool.html

```diff
@@ -330,6 +331,18 @@
                          return 1;
                         return 0; //default return value (no sorting)
                     });
+                    // email
+                    $(data.inputs).each(function (index, input) {
+                        if (email_pattern.test(input.name)) {
```

Comment:
my idea is not to change the HPC, data and email input but rather the other inputs just to be able to sort them.

But if you have a *much* better way to sort the inputs to provide the more important first and the rest after (since we *do not have any thing* for making some tabs for advanced params or inputs)

Reply:
What I think is that if we use the numbers prefix in all inputs, we let the application developer choose which ones are more important than others, without forcing to put always hpc and datasets first.

Reply:
This part is only a demonstrator... But there is an issue with that indexing.
This does not scale at all. I mean, if we have tens of inputs, it's really cumbersome to define and maintain the order...

It will be more flexible if we had input with an integer index key and a category (eventually ith an index also). This way we can print inputs by category (eventually sorted with an integer index) and within each category to sort inputs by an integer index...

What do you think?

Reply:
As in the portal we are ordering them in alphabetic order, I think that everything could be done with numbers. For example, let's say we use for digits, I'd resolve the problem like this:
* category one (let's say we plan in this category to have at least 30 inputs in here): 00[0-2]X
    * If we at first have for example three inputs, we could do something like this to give space between them for new ones:
        * first: 0005
        * second: 00015
        * third: 0025
* category two (ten inputs at most): 003X
* category three (many inputs, but we don't care the order between them: 0040 (same number, they will use the alphabetic order).

This way I think it is not going to be difficult to manage the order of many inputs.